servers/visual/shader_language.cpp

```diff
@@ -3272,15 +3272,17 @@ ShaderLanguage::Node *ShaderLanguage::_parse_expression(BlockNode *p_block, cons
 					case TYPE_IVEC2:
 					case TYPE_UVEC2:
 					case TYPE_MAT2:
-						if (index->type == Node::TYPE_CONSTANT) {
-							uint32_t index_constant = static_cast<ConstantNode *>(index)->values[0].uint;
-							if (index_constant >= 2) {
-								_set_error("Index out of range (0-1)");
+						if (index->type != Node::TYPE_VARIABLE && index->type != Node::TYPE_ARRAY && index->type != Node::TYPE_OPERATOR) {
```

Comment:
Seems a bit convoluted, you could just put the negation of this as `else if` instead of the `else`.

The error should also be improved as it's not longer correct?

Reply:
Changed. Do it looks better now?